pennylane/decomposition/resources.py

```diff
@@ -290,10 +291,18 @@ def my_decomp(wires):
         base_rep = resource_rep(params["base_class"], **params["base_params"])
         params["base_class"] = base_rep.op_type
         params["base_params"] = base_rep.params
+    if op_type is qml.ops.op_math.Prod:
+        resources = defaultdict(int)
+        for rep, count in params["resources"].items():
+            addition = rep.params["resources"] if rep.op_type is qml.ops.op_math.Prod else {rep: 1}
```

Comment:
This seems to handle nested `Prod`, but it only flattens it by one level. What if there's a `Prod` nested in a `Prod` nested in a `Prod`, as in, one of your `sub_rep` is yet again a `Prod`?

Reply:
Ah I see, that makes sense.